### PR TITLE
update veh enums + rage vector2 + cWeaponInfo

### PR DESCRIPTION
**Pull request by altforfr**
kinda unfinished but mostly tested and should all be correct

### CODE REVIEW

**Comment by Yimura** [comment]
I can take a look if you want.

**Comment by altforfr** [comment]
would be appreciated.

File: weapon/CWeaponInfo.hpp [hunk 1]
```diff
@@ -209,15 +210,60 @@ class CWeaponInfo : public CItemInfo
...
 	float m_lock_on_range; //0x0288
 	float m_weapon_range; //0x028C
-	char pad_0290[8]; //0x0290
+	float m_ai_sound_range; //0x0290
+	float m_ai_potential_blast_event_range; //0x0290
 	float m_damage_fall_off_range_min; //0x0298
 	float m_damage_fall_off_range_max; //0x029C
 	float m_damage_fall_off_modifier; //0x02A0
-	char pad_02A4[64]; //0x02A4
+	uint32_t m_vehicle_weapon_hash; //0x02B4
```
Comment (Yimura) [inline weapon/CWeaponInfo.hpp:218]: Jump from 0x2A0 to 0x2B4 is too large.

0x2B4 should be 0x2A4.

The final size should be 0x5E8.

**Review by Yimura — CHANGES_REQUESTED** [review]

**Comment by altforfr** [comment]
thanks :D

**Review by Yimura — APPROVED** [review]

**Pull request merged**